splink/blocking.py

```diff
@@ -24,13 +25,14 @@ def blocking_rule_to_obj(br):
         if blocking_rule is None:
             raise ValueError("No blocking rule submitted...")
         sqlglot_dialect = br.get("sql_dialect", None)
-        salting_partitions = br.get("salting_partitions", 1)
 
-        return BlockingRule(
-            blocking_rule,
-            salting_partitions,
-            sqlglot_dialect,
-        )
+        salting_partitions = br.get("salting_partitions", None)
+        if salting_partitions is None:
+            return BlockingRule(blocking_rule, sqlglot_dialect)
+        else:
+            return SaltedBlockingRule(
+                blocking_rule, sqlglot_dialect, salting_partitions
+            )
```

Comment:
Again, do we not also need to worry about the sql_dialect?

Reply:
Yep. Solved by the protection now built into the `SaltedBlockingRule` class itself in https://github.com/moj-analytical-services/splink/pull/1701/commits/b49bb06cf0564f19eb38b09fb5b78a3233a86c01